File: app/src/main/java/com/itachi1706/busarrivalsg/ListAllBusStopsActivity.java
package com.itachi1706.busarrivalsg;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.itachi1706.busarrivalsg.database.BusStopsDB;
import com.itachi1706.busarrivalsg.recyclerviews.BusStopRecyclerAdapter;
import com.itachi1706.busarrivalsg.objects.gson.ltasg.BusStopJSON;

import java.util.List;

public class ListAllBusStopsActivity extends AppCompatActivity {

    TextView count;
    RecyclerView busStops;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_all_bus_stops);

        count = findViewById(R.id.tvCount);
        busStops = findViewById(R.id.rvAllBusStops);
        if (busStops != null) busStops.setHasFixedSize(true);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        busStops.setLayoutManager(linearLayoutManager);
        busStops.setItemAnimator(new DefaultItemAnimator());

        BusStopsDB db = new BusStopsDB(this);
        List<BusStopJSON> data = db.getAllBusStops();
        count.setText(getString(R.string.label_bus_stops_count, data.size()));

        BusStopRecyclerAdapter view = new BusStopRecyclerAdapter(data);
        busStops.setAdapter(view);
    }
}

File: app/src/main/java/com/itachi1706/busarrivalsg/ListAllBusStopsActivity.kt
package com.itachi1706.busarrivalsg

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import com.itachi1706.busarrivalsg.adapters.BusStopRecyclerAdapter
import com.itachi1706.busarrivalsg.database.BusStopsDb
import com.itachi1706.busarrivalsg.databinding.ActivityListAllBusStopsBinding
import com.itachi1706.helperlib.helpers.EdgeToEdgeHelper

class ListAllBusStopsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityListAllBusStopsBinding.inflate(layoutInflater)
        EdgeToEdgeHelper.setEdgeToEdgeWithContentView(binding.root, this)

        setSupportActionBar(binding.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        binding.rvAllBusStops.setHasFixedSize(true)
        val llm = LinearLayoutManager(this)
        llm.orientation = LinearLayoutManager.VERTICAL
        binding.rvAllBusStops.layoutManager = llm
        binding.rvAllBusStops.itemAnimator = DefaultItemAnimator()

        val db = BusStopsDb(this)
        val data = db.getAllBusStops()
        binding.tvCount.text = getString(R.string.label_bus_stops_count, data.size)

        val view = BusStopRecyclerAdapter(data.toMutableList())
        binding.rvAllBusStops.adapter = view
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}